File: src/replacements.js
// Apply URL replacements
export function applyUrlReplacements(body, config) {
  let updatedBody = body;

  if (config.urlReplacements) {
    for (const [from, to] of Object.entries(config.urlReplacements)) {
      updatedBody = updatedBody.replace(
        new RegExp(escapeRegExp(from), "g"),
        to
      );
    }
  }

  return updatedBody;
}

// Apply PR number replacement
export function applyPRNumberReplacement(body, prNumber) {
  return body.replace(/\${prNumber}/g, prNumber);
}

// Escape special regex characters
function escapeRegExp(string) {
  return string.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
}

File: internal/replacements/replacements.go
package replacements

import (
	"fmt"
	"regexp"

	"github.com/maastrich/gh-pr-number/internal/config"
)

// ApplyURLReplacements applies URL replacements from config
func ApplyURLReplacements(body string, cfg *config.Config, prNumber string) string {
	updatedBody := body

	if cfg.URLReplacements != nil {
		fmt.Println(cfg.URLReplacements)
		for from, to := range cfg.URLReplacements {
			escapedFrom := escapeRegExp(from)
			re := regexp.MustCompile(escapedFrom)
			updatedBody = re.ReplaceAllString(updatedBody, ApplyPRNumberReplacement(to, prNumber))
		}
	}

	return updatedBody
}

// ApplyPRNumberReplacement replaces ${prNumber} placeholders with the actual PR number
func ApplyPRNumberReplacement(body, prNumber string) string {
	re := regexp.MustCompile(`\$\{prNumber\}`)
	return re.ReplaceAllString(body, prNumber)
}

// escapeRegExp escapes special regex characters
func escapeRegExp(s string) string {
	re := regexp.MustCompile(`[.*+?^${}()|[\]\\]`)
	return re.ReplaceAllString(s, `\$0`)
}
